### ISSUE TITLE
slack-desktop 3.2.1-4 fails to install

### ISSUE THREAD

**Issue by ccharles** [aurapm/aura#500]
[`slack-desktop`][1] version 3.2.1-4 [bundles `glibc` 2.27 with Slack][2], apparently due to an incompatibility between Electron and `glibc` 2.28.

Aura refuses to install the new Slack package, thinking that `glibc` will be downgraded:

```
aura >>= Building `slack-desktop`...
==> Making package: slack-desktop 3.2.1-4 (Mon 13 Aug 2018 08:39:42 AM EDT)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading slack-desktop-3.2.1-amd64.deb...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 55.1M  100 55.1M    0     0   9.9M      0  0:00:05  0:00:05 --:--:-- 10.3M
  -> Found slack-desktop.patch
  -> Downloading glibc-2.27-3-x86_64.pkg.tar.xz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 9293k  100 9293k    0     0  3474k      0  0:00:02  0:00:02 --:--:-- 3472k
==> Validating source files with sha256sums...
    slack-desktop-3.2.1-amd64.deb ... Passed
    slack-desktop.patch ... Passed
    glibc-2.27-3-x86_64.pkg.tar.xz ... Passed
==> Extracting sources...
==> Entering fakeroot environment...
==> Starting package()...
patching file usr/share/applications/slack.desktop
==> Tidying install...
  -> Removing libtool files...
  -> Purging unwanted files...
  -> Compressing man and info pages...
==> Checking for packaging issues...
==> Creating package "slack-desktop"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Leaving fakeroot environment.
==> Finished making: slack-desktop 3.2.1-4 (Mon 13 Aug 2018 08:41:59 AM EDT)
loading packages...
warning: downgrading package glibc (2.28-1 => 2.27-3)
resolving dependencies...
looking for conflicting packages...
error: failed to prepare transaction (could not satisfy dependencies)
:: installing glibc (2.27-3) breaks dependency 'glibc=2.28' required by lib32-glibc
```

In contrast, `makepkg -si` works fine:

```
==> Making package: slack-desktop 3.2.1-4 (Mon 13 Aug 2018 09:31:08 AM EDT)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading slack-desktop-3.2.1-amd64.deb...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 55.1M  100 55.1M    0     0   9.9M      0  0:00:05  0:00:05 --:--:-- 10.4M
  -> Found slack-desktop.patch
  -> Found glibc-2.27-3-x86_64.pkg.tar.xz
==> Validating source files with sha256sums...
    slack-desktop-3.2.1-amd64.deb ... Passed
    slack-desktop.patch ... Passed
    glibc-2.27-3-x86_64.pkg.tar.xz ... Passed
==> Extracting sources...
==> Removing existing $pkgdir/ directory...
==> Entering fakeroot environment...
==> Starting package()...
patching file usr/share/applications/slack.desktop
==> Tidying install...
  -> Removing libtool files...
  -> Purging unwanted files...
  -> Compressing man and info pages...
==> Checking for packaging issues...
==> Creating package "slack-desktop"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Leaving fakeroot environment.
==> Finished making: slack-desktop 3.2.1-4 (Mon 13 Aug 2018 09:33:22 AM EDT)
==> Installing package slack-desktop with pacman -U...
loading packages...
warning: slack-desktop-3.2.1-4 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) slack-desktop-3.2.1-4

Total Installed Size:  230.56 MiB
Net Upgrade Size:        0.00 MiB

:: Proceed with installation? [Y/n] 
(1/1) checking keys in keyring                                                                                                                  [########################################################################################] 100%
(1/1) checking package integrity                                                                                                                [########################################################################################] 100%
(1/1) loading package files                                                                                                                     [########################################################################################] 100%
(1/1) checking for file conflicts                                                                                                               [########################################################################################] 100%
(1/1) checking available disk space                                                                                                             [########################################################################################] 100%
:: Processing package changes...
(1/1) reinstalling slack-desktop                                                                                                                [########################################################################################] 100%
:: Running post-transaction hooks...
(1/2) Arming ConditionNeedsUpdate...
(2/2) Updating the desktop file MIME type cache...
```

I think the issue is that the build puts `glibc-2.27-3-x86_64.pkg.tar.xz` in the working directory which Aura then attempts to install.

[1]: https://aur.archlinux.org/packages/slack-desktop/
[2]: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=slack-desktop#n44

**Comment by fosskers** [comment]
> I think the issue is that the build puts glibc-2.27-3-x86_64.pkg.tar.xz in the working directory which Aura then attempts to install.

This is precisely right. Let me check on `master`, this might have been fixed already.

**Comment by fosskers** [comment]
`slack-desktop` builds for me on `master`, which will be released as Aura 2.0 in the near future (I'm cleaning up these last remaining issues and writing release notes).

**Comment by ccharles** [comment]
Great, thanks for the quick response!

I look forward to trying Aura 2.0 when it's released. In the meantime I'm happy to install and upgrade Slack with `makepkg` like a caveman :slightly_smiling_face:.
